### PR TITLE
560 xp add sc manager

### PR DESCRIPTION
**Pull request by gregLibert**

### CODE REVIEW

File: packages/massa-web3/src/experimental/smartContract.ts [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        fee: opts?.fee,
+        periodToLive: opts?.periodToLive,
+        coins: opts?.coins,
+        maxGas: opts?.maxGas,
+        datastore,
+      }
+    )
+
+    const event = opts.waitForSpeculativeExecution
+      ? await operation.getSpeculativeEvents()
```
Comment (AurelienFT) [inline packages/massa-web3/src/experimental/smartContract.ts:212]: I find the boolean misleading. When I read it I think waiting for speculative or not waiting at all. I would prefer something like an enum with no wait, wait speculative, final. If we just want the last two options we can rename it `waitFinalExecution` if false : speculative, if true final

**Review by AurelienFT — COMMENTED** [review]
It's hard to review this whole PR given the fact that most of the changes are reflected on #559

File: packages/massa-web3/src/experimental/smartContract.ts (hunk 1 above)
Comment (gregLibert) [reply packages/massa-web3/src/experimental/smartContract.ts:212]: Is the noWait a real use case?

Comment (gregLibert) [reply packages/massa-web3/src/experimental/smartContract.ts:212]: Let's go with renaming then.

**Review by peterjah — APPROVED** [review]

**Pull request merged**